File: app/src/main/java/com/purpletealabs/bitcoinapp/datasource/ICurrencyDataSource.java
package com.purpletealabs.bitcoinapp.datasource;

import com.purpletealabs.bitcoinapp.dtos.Currency;

import java.util.List;

public interface ICurrencyDataSource {
    void cancel();
    interface Callback {
        void getCurrenciesFailure();

        void getCurrenciesResult(List<Currency> countries);
    }

    void getSupportedCurrencies(Callback callback);
}

File: app/src/main/java/com/purpletealabs/bitcoinapp/datasource/ICurrencyDataSource.kt
package com.purpletealabs.bitcoinapp.datasource

import com.purpletealabs.bitcoinapp.dtos.Currency

interface ICurrencyDataSource {
    fun cancel()
    interface Callback {
        fun getCurrenciesFailure()

        fun getCurrenciesResult(countries: List<Currency>)
    }

    fun getSupportedCurrencies(callback: Callback)
}